### DISCUSSION TITLE
Is it intended that exiting animation happens after layout recalculation?

### DISCUSSION THREAD

**Discussion by vjeranc** [software-mansion/react-native-reanimated#3596] [Q&A] [answered]
I noticed that exiting animation starts after the layout reflects the removal of the element.

1. component is mounted
2. layout is recalculated
3. `entering` animation is triggered
4. component is dismounted
5. layout is recalculated
6. `exiting` animation is triggered

I would expect 4. 5. 6. to be reversed (first exiting animates, then component dismounts and then layout recalculates - 6. 5. 4 or 4. 6. 5.).
There are cases where elements start to occupy space of the removed element before it actually animates away from the screen. This can create visual overlap and make the removal look buggy.

**Comment by chrisbruford** [comment]
@vjeranc I don't suppose you ever solved this? I'm having the same problem

**Comment by MatiPl01** [comment, marked as answer]
Layout animations don't have an impact on when components are mounted/unmounted. They just animate the layout change of the specific component (the one on which they are used).

If you remove the component, all content rendered below will shift up immediately, unless you pass the `layout` property to every sibling of the removed component rendered below.

The solution I proposed in [this](https://github.com/software-mansion/react-native-reanimated/discussions/5857#discussioncomment-9992457) discussion might be helpful for you.

**Reply by MatiPl01** [reply]
@vjeranc @chrisbruford I will mark this comment as answer. The behavior you expect is not how layout animations are designed to work and the actual behavior is intended. If you want all other rendered content to respond to layout changes, use `useAnimatedStyle` to animate the component (or dimensions of its parent).

Feel free to ask if you don't know how to implement the animation properly.
